api/v1beta1/types_class.go

```diff
@@ -235,6 +235,13 @@ type AzureManagedControlPlaneClassSpec struct {
 	// SecurityProfile defines the security profile for cluster.
 	// +optional
 	SecurityProfile *ManagedClusterSecurityProfile `json:"securityProfile,omitempty"`
+
+	// ASOManagedClusterPatches defines JSON merge patches to be applied to the generated ASO ManagedCluster resource.
+	// WARNING: This is meant to be used sparingly to enable features for development and testing that are not
+	// otherwise represented in the CAPZ API. Misconfiguration that conflicts with CAPZ's normal mode of
+	// operation is possible.
```

Comment:
@dtzar Feel free to offer feedback on how scary this message should be. I also tried to reflect this sentiment in the release note in the PR description.

Reply:
LGTM